persistence-modules/spring-boot-persistence-5/src/test/java/com/baeldung/transactional/TestOrderServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package com.baeldung.transactional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest(classes = { Application.class, TestOrderServiceTest.TestOrderService.class, TestOrderRepository.class })
+class TestOrderServiceTest {
+
+    @Autowired
+    private TestOrderService underTest;
+
+    @Autowired
+    private TestOrderRepository testOrderRepository;
+
+    @Service
+    static class TestOrderService {
+
+        @Autowired
+        private TestOrderRepository repository;
+
+        @Transactional
+        public void createOrderPublic(TestOrder order) {
+            repository.save(order);
+            throw new RuntimeException("Rollback createOrderPublic");
+        }
+
+        @Transactional
+        void createOrderPackagePrivate(TestOrder order) {
+            repository.save(order);
+            throw new RuntimeException("Rollback createOrderPackagePrivate");
+        }
+
+        @Transactional
+        protected void createOrderProtected(TestOrder order) {
+            repository.save(order);
+            throw new RuntimeException("Rollback createOrderProtected");
+        }
+
+        @Transactional
+        private void createOrderPrivate(TestOrder order) {
+            repository.save(order);
+            throw new RuntimeException("Rollback createOrderPrivate");
+        }
+
+        public void callPrivate(TestOrder order) {
+            createOrderPrivate(order);
+        }
+    }
+
+    @AfterEach
+    void afterEach() {
+        testOrderRepository.deleteAll();
+    }
+
+    @Test
+    void givenPublicTransactionalMethod_whenCallingIt_thenShouldRollbackOnException() {
+        assertThat(testOrderRepository.findAll()).isEmpty();
```

Comment:
```suggestion
        assertThat(testOrderRepository.findAll())
            .isEmpty();
```
Let's format the chained method calls on separate lines to make them more visible

Reply:
Is this really the right way? I apply the Baeldung code style which, after I added line break like you suggested, corrected it back to the way it is now.

Reply:
@LeoHelfferich from the author guidelines:

> formatting - fluent APIs
> - **sometimes, the default formatter does need a bit of manual help**
> - for example, fluent APIs are more readable if we split each method call on its own line
> - if the method call is small enough, and very logically connected to the next one - you can also group 2 on a single line